spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/util/SpanIdGenerator.java

```diff
@@ -0,0 +1,11 @@
+package org.springframework.cloud.sleuth.util;
+
+/**
+ * Contract for generating id for spans and traces
+ *
+ * @author Marcin Grzejszczak
+ */
+public interface SpanIdGenerator {
```

Comment:
why? This just opens up the implementation for someone to use something that's not a random generator. I'd remove this type (as discussed earlier) and enforce random. Hooks are easier to add than take away.